### ISSUE TITLE
blackbox_render issue with large timestamps

### ISSUE THREAD

**Issue by kaefert** [cleanflight/blackbox-tools#12]
I just tried the luxfloat PID controller and the blackbox-log of that flight can be decoded just fine, but the rendering shows a unchanging PID table, unchanging command inputs and missing all the lines of motor outputs and gyro.

blackbox_decode works just fine and produces a csv document that looks all right.

```
kaefert@mint ~/Downloads/quadcopter_stuff/blackbox-log/2015-08-09_1110 try luxfloat PID $ ~/Downloads/quadcopter_stuff/blackbox-log/blackbox-tools/obj/black
box_decode LOG00138.TXT 
Decoding log 'LOG00138.TXT' to 'LOG00138.01.csv'...

Log 1 of 1, start 43:28.366, end 45:53.195, duration 02:24.828

Statistics
Looptime           3504 avg           28.3 std dev (0.8%)
I frames    1292   51.2 bytes avg    66199 bytes total
P frames   40029   29.4 bytes avg  1175762 bytes total
H frames      12   11.0 bytes avg      132 bytes total
G frames     915   11.5 bytes avg    10523 bytes total
E frames       2    9.5 bytes avg       19 bytes total
Frames     41321   30.1 bytes avg  1241961 bytes total
Data rate  285Hz   8659 bytes/s      86600 baud

1 frames failed to decode, rendering 4 loop iterations unreadable. 6 iterations are missing in total (21ms, 0.01%)

kaefert@mint ~/Downloads/quadcopter_stuff/blackbox-log/2015-08-09_1110 try luxfloat PID $ ~/Downloads/quadcopter_stuff/blackbox-log/blackbox-tools/obj/black
box_render LOG00138.TXT 
4355 frames to be rendered at 30 FPS [2:26]

Rendered 500 frames (11.5%)...
Rendered 1000 frames (23.0%)...
Rendered 1500 frames (34.4%)...
Rendered 2000 frames (45.9%)...
Rendered 2500 frames (57.4%)...
Rendered 3000 frames (68.9%)...
Rendered 3500 frames (80.4%)...
Rendered 4000 frames (91.8%)...
Rendered 4355 frames (100.0%).
```

**Comment by kaefert** [comment]
oh... sorry, I simply tried again (a new flight also with luxfloat pid controller) and this time both decode and render tools worked. but it's still strange though that with my first luxfloat try decode worked but render delivered such a strange result although the decoded csv file looks like all the infos are there.

If one of the devs want's to take a look at that log, I can send it to you in a pm.

**Comment by thenickdude** [comment]
Please email me the log at n.sherlock@gmail.com

**Comment by kaefert** [comment]
mail sent. hope you find time to take a look and maybe find the reason for the strange behaviour :)

**Comment by thenickdude** [comment]
Ah I think the issue is that the timestamps are so large that they wrap around and end up looking negative. At least in the [web viewer](https://github.com/cleanflight/blackbox-log-viewer) this is the case:

![screen-shot-2015-08-19-at-2 51 41-pm](https://cloud.githubusercontent.com/assets/1921411/9347757/f61755a2-4681-11e5-99d0-17aa5ee580b1.jpg)

It looks like your flight didn't start until 43 minutes after your flight controller was powered, is that accurate?

As an aside, your P and D are wayyy too high, which is causing those incredible oscillations on the motor commands (you can see that the gyro hardly moves at all, but the P and D lines are making huge overreactions).

**Comment by kaefert** [comment]
ahh! yes that's possible. I often first power the flight controller (+gps module powered by the 3.3V output of my flight controller) with an usb cable to get a gps lock. Once that's done which sometimes takes a while (especially with that build where the gps antenna was looking down because of https://github.com/cleanflight/cleanflight/issues/1183 ) I connect the lipo before removing the usb cable to not loose the gps lock.

I guess it's possible I lost track of time reading some webpages or something and only started flying 43 minutes after first connecting the usb cable.

And yes I definitely did not yet get the hang of PID tuning... I'll need to get back to trying to learn that, at the moment my quad is pulled down and is in the progress of beeing rebuilt into a new frame that I designed and printed http://www.thingiverse.com/thing:974479

I think the PIDs in this flight where the defaults from the cleanflight configurator for the luxfloat PID-controller.. I've later tried to reduce the PIDs with the luxfloat controller but then in felt really heavy and sluggish.

I've mostly flown with the default PID-controller with default PIDs until now, which from the feeling of it was quite fine, though the blackbox overlay shows ugly oscilations on the motor outputs: https://www.youtube.com/watch?v=iRjbB9s6cwM

**Comment by thenickdude** [comment]
Okay great, thanks! I'll see about fixing the timestamp display. It will be more difficult to display powerup times in excess of 71 minutes, but I should be able to get >35 minute recordings to display properly without any drama.

**Comment by thenickdude** [comment]
This should now be fixed in [blackbox_render v0.4.2](https://github.com/cleanflight/blackbox-tools/releases) and the latest [web log viewer](https://github.com/cleanflight/blackbox-log-viewer), thanks for the log!
